Mage.Sets/src/mage/cards/g/GaleWaterdeepProdigy.java

```diff
@@ -0,0 +1,98 @@
+package mage.cards.g;
+
+import mage.MageInt;
+import mage.abilities.common.ChooseABackgroundAbility;
+import mage.abilities.common.SpellCastControllerTriggeredAbility;
+import mage.abilities.effects.CastCardFromGraveyardThenExileItEffect;
+import mage.abilities.effects.Effect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.filter.FilterCard;
+import mage.filter.FilterSpell;
+import mage.filter.common.FilterInstantOrSorcerySpell;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+import mage.target.common.TargetCardInYourGraveyard;
+import mage.watchers.common.CastFromHandWatcher;
+
+import java.util.UUID;
+
+/**
+ * @author Rjayz
+ */
+public final class GaleWaterdeepProdigy extends CardImpl {
+
+    public GaleWaterdeepProdigy(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+        
+        // Whenever you cast an instant or sorcery spell from your hand, you may cast up to one of the other type from your graveyard. 
+        // If a spell cast from your graveyard this way would be put into your graveyard, exile it instead.
+        CastCardFromGraveyardThenExileItEffect effect = new CastCardFromGraveyardThenExileItEffect();
+        effect.setText("you may cast up to one of the other type from your graveyard. If a spell cast from your graveyard this way would be put into your graveyard, exile it instead.");
+        this.addAbility(new GaleWaterdeepProdigyTriggeredAbility(effect,
+                new FilterInstantOrSorcerySpell("an instant or sorcery spell from your hand"),
+                false), new CastFromHandWatcher());
```

Comment:
Can you please organize the indenting in here to make it easier to see which parameter goes to which method/constructor.

Reply:
I agree it's not very clear, I have changed the indentation to be more clear